rpcserver.go

```diff
@@ -792,6 +792,22 @@ func (r *rpcServer) SendCoins(ctx context.Context,
 		in.Addr, btcutil.Amount(in.Amount), int64(feePerKw),
 		in.SendAll)
 
+	// Decode the address receiving the coins, we need to check whether the
+	// address is valid for this network.
+	addr, err := btcutil.DecodeAddress(
+		in.Addr, activeNetParams.Params,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make the check on the decoded address according to the active network.
+	if !addr.IsForNet(activeNetParams.Params) {
+		return nil, fmt.Errorf("address: %v is not valid for this network: %v",
+			addr.String(), activeNetParams.Params.Name,
```

Comment:
We should normalize these names as regtest, for example, will return "testnet3".

Reply:
I saw the other PR, it has a function:

```
// activeNetParams returns a string representation of the current active
// network.
func activeNetName() (string, error) {
	switch {
	case cfg.Bitcoin.MainNet || cfg.Litecoin.MainNet:
		return "mainnet", nil
	case cfg.Bitcoin.TestNet3 || cfg.Litecoin.TestNet3:
		return "testnet", nil
	case cfg.Bitcoin.SimNet || cfg.Litecoin.SimNet:
		return "simnet", nil
	case cfg.Bitcoin.RegTest || cfg.Litecoin.RegTest:
		return "regest", nil
	default:
		return "", fmt.Errorf("unknown network for addr")
	}
}
```

Is this what you meant = normalize an input from ("testnet3") -> ("testnet")?

Reply:
Yep!

Reply:
@wpaulino Thanks! I basically just reapplied roasbeefs changes including refactoring duplicate use of DecodeAddress